evm_loader/program/src/evm/buffer.rs

```diff
@@ -83,6 +83,11 @@ impl Buffer {
         Buffer::new(Inner::Empty)
     }
 
+    #[must_use]
+    pub fn buffer_is_empty(&self) -> bool {
+        matches!(self.inner, Inner::Empty)
+    }
```

Comment:
This is not needed because of:
```rust
impl Deref for Buffer {
    type Target = [u8];

    #[inline]
    fn deref(&self) -> &Self::Target {
        debug_assert!(!self.ptr.is_null());

        unsafe { std::slice::from_raw_parts(self.ptr, self.len) }
    }
}
```

So you can write `buffer.is_empty()` instead of `buffer.buffer_is_empty()`.

Reply:
In that case, why do we have the `Empty` variant at all? We could just store `Owned(Vec::default)` instead.

Reply:
I assumed `Empty` had some sort of special meaning.